core/src/blocktree.rs

```diff
@@ -468,29 +461,99 @@ impl Blocktree {
 
         writebatch.put_bytes::<cf::Coding>((slot, index), bytes)?;
 
+        if let Some((data, coding)) =
+            self.try_erasure_recover(&erasure_meta, slot, &HashMap::new(), Some((index, bytes)))?
+        {
+            let mut erasure_meta_working_set = HashMap::new();
+            erasure_meta_working_set.insert((slot, set_index), erasure_meta);
+
+            self.insert_data_blob_batch(
+                &data[..],
+                &mut HashMap::new(),
+                &mut erasure_meta_working_set,
+                &mut HashMap::new(),
+                &mut writebatch,
+            )?;
+
+            erasure_meta = *erasure_meta_working_set.values().next().unwrap();
+
+            for coding_blob in coding {
+                erasure_meta.set_coding_present(coding_blob.index(), true);
+
+                writebatch.put_bytes::<cf::Coding>(
+                    (coding_blob.slot(), coding_blob.index()),
+                    &coding_blob.data[..BLOB_HEADER_SIZE + coding_blob.size()],
+                )?;
+            }
+        }
+
         writebatch.put::<cf::ErasureMeta>((slot, set_index), &erasure_meta)?;
 
         self.db.write(writebatch)?;
 
-        self.try_erasure_recover(&erasure_meta, slot, set_index)
+        Ok(())
     }
 
     fn try_erasure_recover(
         &self,
         erasure_meta: &ErasureMeta,
         slot: u64,
-        set_index: u64,
-    ) -> Result<()> {
-        match erasure_meta.status() {
+        prev_inserted_blob_datas: &HashMap<(u64, u64), &[u8]>,
+        new_coding_blob: Option<(u64, &[u8])>,
+    ) -> Result<Option<(Vec<Blob>, Vec<Blob>)>> {
+        let blobs = match erasure_meta.status() {
             ErasureMetaStatus::CanRecover => {
-                let recovered = self.recover(slot, set_index)?;
+                let (data, coding) = self.recover(
+                    slot,
+                    erasure_meta,
+                    prev_inserted_blob_datas,
+                    new_coding_blob,
+                )?;
+                let recovered = data.len() + coding.len();
+                info!("recovered {} blobs", recovered);
                 inc_new_counter_info!("blocktree-erasure-blobs_recovered", recovered);
```

Comment:
is recover() only returning the blobs for erasures?

Reply:
Yes, it only returns the reconstructed ones and not the whole set